cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageAdmin.java

```diff
@@ -137,18 +140,21 @@ public Lineage computeLineage(final Id.Stream sourceStream, long startMillis, lo
    * @return metadata associated with a run
    */
   public Set<MetadataRecord> getMetadataForRun(Id.Run run) throws NotFoundException {
-    entityExistenceVerifier.ensureExists(run.toEntityId());
+    ProgramRunId programRunId = run.toEntityId();
+    entityExistenceVerifier.ensureExists(programRunId);
+
+    // TODO: Remove this once lineageStoreReader stops using deprecated Id class
+    Set<NamespacedEntityId> runEntities = new HashSet<>(lineageStoreReader.getEntitiesForRun(run));
```

Comment:
My bad, outdated comment before I thought I wasn't going to change `getEntitiesForFun`